### PR TITLE
[TEST-MERGE FIRST] Allows all limbs to be dismembered and significantly refactors wounds

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request
**_THIS PR UPDATES THE SCAR VERSION - ALL EXISTING SCARS WILL BE WIPED_**

Expands the wound system functionality to support any type of limb at all.

To do this, wounds have been significantly refactored. For starters, wounds now use limb biotype instead of wound type for determining what they can be applied to. They also use singleton instances for most "can we apply this" checks instead if copy pasted initial(). 

Wounds now use a "wound series" instead of wound_type for determining the, well, series. Previously, all WOUND_BLUNT wounds were considered bone wounds, making it impossible to have multiple WOUND_BLUNT wounds at once. Now, its based on wound series - bone wounds are of the blunt bone wound series, and use the typical logic. 
One change that results from this is the ability for everything with a jointed limb to get a dislocation. Yes, this includes things like prosthetics.

On the note of external and internal biotypes: Exterior are bones, Interior is flesh. Interior protects exterior from slash until its mangled, at which point it either exposes exterior to slash or allows dismemberment if theres no exterior.
Basically - it acts the exact same way, except its not hardcoded, and its more modular.

A lot, lot more changes were made - I cant name them all, but if youre interested, you can read up. Wounds have more procs, more modularization, and less hardcoding.

Sadly, scars have been updated in such a way so that the wound version must be updated. Scars will be deleted.

## Why It's Good For The Game

As it stands, half the limbs in the game can't be dismembered. This changes that, allowing every single limb to be dismembered.

The two dismemberment critera are now:

1. If able to get mangled flesh or bone, it can be dismembered once it gets mangled flesh and bone (or JUST flesh if it only has a internal biostate, vice vers afor bone if external only)
2. If it cant be dismembered by the above, it will have a chance to dismember when at or above 80% of its total max damage

Finally, code being better is usually a good thing.

## Changelog
:cl:
balance: Prosthetics and slimepeople can now have limbs dismembered
balance: Slimepeople can now receive slash wounds, but cannot bleed
balance: Most limbs can now be dislocated
refactor: Scar backend reworked, scars will be wiped as they update to the new format
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
only open to see the diff right now but its near feature completion

**Comment by carlarctg** [comment]
Cool! This is gonna open up some neat depth for prosthetics and augments. 

It'd be fantastic if the massive limb disabling vulnerability of surplus prosthetics was replaced with a massive wound penalty on them. MUCH more interesting that way.

**Comment by AtomTheProphet** [comment]
> Cool! This is gonna open up some neat depth for prosthetics and augments.
> 
> It'd be fantastic if the massive limb disabling vulnerability of surplus prosthetics was replaced with a massive wound penalty on them. MUCH more interesting that way.

I'd imagine it would cause primarily electrical damage and maybe a bit of overheating.

**Comment by Serxule** [comment]
synthetic/augment is already balanced, slightly tougher with some upsides but entirely fucked by 1 emp, this is stupid to make them a human reskin

**Comment by mc-oofert** [comment]
chaplain synths get critted by 1 EMP shot and killed by 2

**Comment by SimplySmiley123** [comment]
> > synthetic/augment is already balanced, slightly tougher with some upsides but entirely fucked by 1 emp, this is stupid to make them a human reskin
> 
> Being better except for the one counter is not balance. This is not a game with pick counterpick meta, you won't see EMP near often enough for that weakness to matter. Don't know if this change is good (literally no opinion) but i disagree with your reasoning against it.

And this doesn't really balance it out either imo.
You still get fucked with one item's singular use. And as a bonus, said item is *extremely* easy to obtain even as a non-antagonist, while antagonists get it for cheap.
And with this you get fucked more for bothering. I bet this will make it so robos get even less people coming over for anything, while mechanical sect chaplains get even less people interested in it.

**Comment by Serxule** [comment]
> > synthetic/augment is already balanced, slightly tougher with some upsides but entirely fucked by 1 emp, this is stupid to make them a human reskin
> 
> Being better except for the one counter is not balance. This is not a game with pick counterpick meta, you won't see EMP near often enough for that weakness to matter. Don't know if this change is good (literally no opinion) but i disagree with your reasoning against it.

then why havent borgs been changed? as it is synthetic humanoids are still weaker than borgs save for equipment

**Comment by mc-oofert** [comment]
> Regardless, EMP damage has been significantly reduced in the last commits, down to 48 on heavy, with the rationale of "the hard counter is far less needed"

i had a round 1 hour ago where someone got instacrit by emp for having augments

**Comment by SomeoneYouProbablyKnow** [comment]
> > > synthetic/augment is already balanced, slightly tougher with some upsides but entirely fucked by 1 emp, this is stupid to make them a human reskin
> > 
> > 
> > Being better except for the one counter is not balance. This is not a game with pick counterpick meta, you won't see EMP near often enough for that weakness to matter. Don't know if this change is good (literally no opinion) but i disagree with your reasoning against it.
> 
> then why havent borgs been changed? as it is synthetic humanoids are still weaker than borgs save for equipment

Borgs are being changed right now?? The double flash PR was merged not a week ago, silicon policy is in the works. Borg combat and behavior are a known issue, it being as it is (bad) doesn't justify something else being bad

**Comment by Ryll-Ryll** [comment]
As I said to you on discord, this seems like a big expansion of complexity that people have to keep in mind for treating entirely parallel sets of wounds to the original set, and also a pretty big increase of code complexity. I still don't think this is a positive improvement.

**Comment by nikothedude** [comment]
> As I said to you on discord, this seems like a big expansion of complexity that people have to keep in mind for treating entirely parallel sets of wounds to the original set, and also a pretty big increase of code complexity. I still don't think this is a positive improvement.
> 
> We don't want people to have to be checking the wiki or their wound analyzer after every step of treatment to try to remember what to do next because there's suddenly twice as much to remember, let alone checking it 5 times before getting treatment to even know what you need to gather.

1. Im generally pretty clear with how to fix a wound, what the wound does, etc. in the descriptions. Yes, people will have to check it, or check the wiki, INITIALLY. HOWEVER. Once people start actually dealing with these wounds, they will start remembering how to treat them. I think it's paranoid to assume people just won't _learn_ to deal with these wounds.
1.1: Also, most robotic wounds wont even crop up that often, since robotic limbs are rare/uncommon and cyborg limbs have inherent damage/wound resist.

2. The code complexity is only in the form of me improving the code to work better or be more modular. Sure, wound code is more complex, but it is what lets us make any limb have any wound, give custom scars, etc.
3. Wounds aside, this PR still holds on the idea that it lets any limb be dismembered. Since dismemberment these days relies on wounds, anything without wounds cant be dismembered, so not giving these limbs wounds will keep them undismemberable, even though there really isnt there _reason_ aside from "we dont want wounds". I _suppose_ we could just give them filler wounds that do nothing but mangle flesh/bone, but that's a bit lame.

**Comment by nikothedude** [comment]
Worst case scenario though, I just atomize the wounds out of this PR, keep everything the way it was, and make this exclusively the code improvements.

**Comment by nikothedude** [comment]
I DID NOT MARK THAT

**Comment by Comxy** [comment]
I don't think slime people should get burn wounds when they are resistent to fire damage. Also I don't think brute wounds on slime people makes much sense since they would just replace the mass they lost

**Comment by nikothedude** [comment]
> We don't want people to have to be checking the wiki or their wound analyzer after every step of treatment to try to remember what to do next because there's suddenly twice as much to remember, let alone checking it 5 times before getting treatment to even know what you need to gather.


Also! I made some effort to make, for at least slash/pierce wounds, wound treatments doable with stereotypical things. You can fix wires with sutures at reduced efficiency, you can use bone gel to secure your limb's internals, you can use cryogenics to treat burns, you can weld a limb closed with a cautery.

**Comment by necromanceranne** [comment]
I was aware for a while that inorganic limbs had somehow gained dismemberment immunity. I was especially aware of this when I had noticed that very rarely did I ever see someone with a prosthesis actually lose that prosthesis, even though prior to wounds, prosthesis limbs were extremely dismemberment prone.

I already brought up a while ago, in [this PR](https://github.com/tgstation/tgstation/pull/71739) that prosthesis limbs and cybernetic limbs had suddenly gained a really weird amount of resistance to some effects that previously they simply didn't have. I see no reason why we can't bridge this gap after a few years of having wounds as they are, only mattering for organics and leaving gaps in weird places like prosthetics. I don't think it is good for the game for people to have negative quirks benefit them via resistance to various lethal afflictions.

It is also very weird how wounds effectively made an upside for slimepeople (limb growth) entirely irrelevant because they simply do not lose limbs enough for it to be a regular occurrence (beyond their limbs melting from blood loss via toxin healing chems). This is where wounds have become a net negative for species via undermining their features. By making wounds relevant again to these species, we return a bit of depth.

Definitely think this is what we need. If we can't somehow manage niche species/limb configurations having wounds, what is the point in wounds if they're inconsistent about how they occur?

**Comment by nikothedude** [comment]
this pr is almost ready, i just need to do some testing

**Comment by Serxule** [comment]
> > Securing Failure: 1:1 to blunt with some caveats: Arm injuries add a chance to drop what youre holding when the arm is hit
> 
> Organic arms should have a chance to drop the item. Mechanical arms should have a chance to lock up, preventing the item from being dropped at all (or, if the hand is empty, preventing anything from being picked up).

incredibly true, mechanical failures (unless there's a failsafe or emergency stop, which only prevents further movement and doesnt fix the problem) are going to generally continue doing what they were doing when they fail or lockup

**Comment by Serxule** [comment]
> 

> * * Intensity slowly raises over time, slowing when gauzed, when on stasis, and when the victim is lying down.

why exactly does gauze... do anything??
> * Blunt
> * * Securing Failure: 1:1 to blunt with some caveats: Arm injuries add a chance to drop what youre holding when the arm is hit, leg injuries add a chance to fall over when hit (These two are subject to change, they arent necessary). On the head, you get camerashake when you move or when the head is hit, along with dizziness. On the chest, you get nausea when you move or are hit (exceptions for if youre buckled to a roller bed and move). On all, slight organ damage when you move or get attacked (also can be changed, not crucial)
> * * Treatment: Screwdriver or percussive mainteance for T1. Screwdriver/Wrench (low chance for success if you dont know robo/engi wires or wear a diag hud, or if you do it on yourself) or bone gel (high success chance but guaranteed ~40 damage and takes a while) for T2 then welding (again, high chance to succeed). T3: Get burn wound (can be done harmlessly with a buddy and a welder) then 1. percussive maintenance 2. aggrograb help intent 3. use plunger to reset superstructure - or just use an RCD. Then, T2 treatment. All levels are subject to gauze. Movement stuff is reduced when walking, resting, or in anti-grav.

this is a fucking MASSIVE and horribly unnecessary nerf, why make it harder for people to heal (especially by themselves which will be 90% of the time) when augments/synthetic is _already rare_?? this is such a massive nerf that would genuinely make any augments a full downgrade.
> * Burn
> * * Overheating: Increases damage taken, in general, more than flesh burns, and are easier to get (Subject to change, tempted to incrase the thresholds).
> * * All burn wounds are subject to temperature. Burn wounds hold a virtual temp var, and equalize both their temp with their victim and their victim temps with themself, which results in quick cooling at high tiers, but also quick overheating of the victim.
> * * When a burn wound goes below x it reduces severity or removes. If it goes above y temp, it GAINS severity.
> * * You can treat these by cryogenics, ideally, as its harmless and destroys bodytemp while insulating from cold. You can, however, use reagent contact to quickly reduce temp, at the cost of causing significant thermal shock in the form of brute damage (Fire extinguishers! Showers!).

augments/synthetic is literally heat/cold/pressure proof. what the fuck?

> Wounds are a very fun addition to medical gameplay and general gameplay, 

no, they arent fun at all, for anybody, they're extremely annoying especially when they are hard to deal with and fix (for anybody) and insanely bad downsides like yours.

**Comment by Dorsisdwarf** [comment]
Pretty weird choices for putting aggressive and debilitating wounds onto limbs that are immune or highly resistant to those damage types.

Also, not sure why anyone would willingly want a robot limb now.

**Comment by necromanceranne** [comment]
Being resistant to damage already grants resistance to wounding. Damage taken AFTER mitigations is the direct contributing factor to wound rolling percentage and severity.

Anyone can become extremely resistant to damage. That's what armor does. Wounding probability for organic limbs currently respect damage resistance in an obtuse way rather than a direct way, and there are ways that organic limbs that can be wounded right now can become resistant to damage on the species/limb level. 

There are no limbs that grant (technical) immunity to damage. Synthetic limbs lost anything of the sort in this PR https://github.com/tgstation/tgstation/pull/76817, and this pr already covers species that should be immune to wounds (like golems, who can get some extreme resistances to things like burn damage even if it isn't technical immunity, it is effective immunity, yet they still haven't lost wound immunity).

**Comment by nikothedude** [comment]
> Being resistant to damage already grants resistance to wounding. Damage taken AFTER mitigations is the direct contributing factor to wound rolling probability and potential severity.
> 
> Anyone can become extremely resistant to damage. That's what armor does. Wounding probability for organic limbs currently respect damage resistance in an obtuse way rather than a direct way. And, there are ways that organic limbs that can be wounded currently can also become resistant to damage on the species/limb level. There is no consistency, as this PR points out.
> 
> There are no limbs that grant (technical) immunity to damage. Synthetic limbs lost anything of the sort in this PR #76817, and this pr already covers species that should be immune to wounds (like golems, who can get some extreme resistances to things like burn damage even if it isn't technical immunity, it is effective immunity, yet they still haven't lost wound immunity).



> > Securing Failure: 1:1 to blunt with some caveats: Arm injuries add a chance to drop what youre holding when the arm is hit
> 
> Organic arms should have a chance to drop the item. Mechanical arms should have a chance to lock up, preventing the item from being dropped at all (or, if the hand is empty, preventing anything from being picked up).

Genuienly didnt know this

It was already on the chopping block, since interaction delayt and limping is a thing, so

**Comment by nikothedude** [comment]
> why exactly does gauze... do anything??

Because it's a thing for every single wound. Having an exception because "it doesnt make sense" is dumb, we can make up any in-universe justification we want. Besides - the more ways we have to slow the intensity of a wound through non-intrusive means (e.g. surgery and shit), the better - ghetto treatments or quick remedies are important to make wounds bearable.

> this is a fucking MASSIVE and horribly unnecessary nerf, it would make combat basically impossible for anyone augments/synthetic, and why make it harder for people to heal "wounds" (especially by themselves which will be 90% of the time) when augments/synthetic is _already rare_?? this is such a massive nerf that would genuinely make any augments a full downgrade.

It would help if you specified what is the horrible and awful downgrade, theres a lot of effects.

> augments/synthetic is literally heat/cold/pressure proof. what the fuck?

Thats just wrong, actually, I mean-they can get burn damage for a reason.
anybody) and insanely bad downsides like yours.

> A lot of these make sense for prosthetics, because prosthetics are meant to be shit and a downgrade, but otherwise? no.

Totally fine with this, but do keep in mind, with the way things are right now, things NEED wounds to resist dismemberment. If a maintainer decrees, I can pump cyborg wound resist to make these wounds significantly harder to get, though I'll probably do a lot of adjusting them anyway.

**Comment by Donglesplonge** [comment]
so i was looking at it and for the tier 3 blunt, do i actually HAVE to have a plunger? plungers are fairly common but if you're already having to go out and get a buddy to help you out with the aggrograb on help (i could just be reading this wrong) then you also have to go grab a plunger too, this isn't like medical wherein their tools are always within arms reach of themselves, roboticists don't use plungers (even if they can print one from the science lathe as cytologists need them), maybe also let them use crowbars to jag under the frame with a reduced success rate? i just feel like "hey cripple go find a plunger we don't carry plungers on ourselves" is gonna happen ALOT.

also kinda torn on surplus prosthetics getting more health, they're supposed to be cheap pieces of shit, and while wounds do suffice for making them a bitch, they shouldn't get a boon in health to compensate for it honestly, i wouldn't know the implications of it from a balance perspective but i feel like surplus prosthetics should be faster to fix than augs and regular robotic limbs as they're stripped down to their bare essentials to function, they're made of cheap ass NT approved pig iron and have their wiring exposed in the frame (don't mind that surpluses don't have any discernable wiring in the sprite), they would have a hell of an easier time say, smacking it back to form, or inserting wiring, but thats just me. 

otherwise looks fun, can't wait to get double EMP'd into oblivion in the crossfire of a heretic fighting security again 👍

**Comment by nikothedude** [comment]
> so i was looking at it and for the tier 3 blunt, do i actually HAVE to have a plunger? plungers are fairly common but if you're already having to go out and get a buddy to help you out with the aggrograb on help (i could just be reading this wrong) then 

nope! A plunger is an alternate option to molding the mallaeble limb with your hands, or using an RCD, or hitting it repeatadly until it works (percussive maintenance is a mechanic). A crowbar is a good tool I neglected, though.

As for prosthetics - a esword delimbs one in 3 hits, now, I seriously think its fair.

**Comment by nikothedude** [comment]
When I'm satisifed with the state of the wounds, I'll upload a few videos showing them off. Effects, treatment, sounds, etc.

**Comment by nikothedude** [comment]
> The exception is fine here because prosthetic limbs are intentionally different than organic ones. Differences in how they're repaired is actually cool, because it highlights that yes, your arm is actually made of metal and not just spraypainted silver.
> 
> Despite using suture and mesh to heal brute and burn damage on every single organic limb, prosthetics use welders and wires. This is because they are indeed different, and the difference fleshes out the game.

True, different treatment is nice,

But having the funtionality of gauze is still, IMO, if not important, at least a nice touch of color on the wounds system. I could replace gauze with something else, but why? Why go through the ffort of refactoring the gauze system to be so much more modular to support this when I can just use gauze? It's not like I can't even find a easy rationale for its use. Blunt: It secures the limb. Slash: It secures the wires and helps with insulation. Burn: It binds the limb and reduces the ability for it to warp outwards when thermally shocked.

**Comment by zxaber** [comment]
Maybe an alternative item where you can use cable cuffs for the same purpose? That way robotics has a way to handle it.

**Comment by Serxule** [comment]
> > this is a fucking MASSIVE and horribly unnecessary nerf, it would make combat basically impossible for anyone augments/synthetic, and why make it harder for people to heal "wounds" (especially by themselves which will be 90% of the time) when augments/synthetic is _already rare_?? this is such a massive nerf that would genuinely make any augments a full downgrade.
> 
> It would help if you specified what is the horrible and awful downgrade, theres a lot of effects.

dropping items when hit/falling down when hit would make combat impossible, itd be the equivalent of being hit by a baton every time you got hit

camera shake, dizziness, and nausea also make combat impossible, by making it incredibly hard to aim/melee/etc

making it hard to fix via self or for anyone else to without 1-2 of 5 specific items is stupid, especially when one of the bigger upsides is that you can literally do perfect self surgery (for basic surgeries like organ manip) on augments/synthetic, and considering most of the time you're going to be healing yourself when augmented (medical is useless and engineering isnt gonna know shit, robotics has a total of 2 slots)

why would bone gel work, at all?? it doesnt even make sense mechanically or in a "realistic" sense, and why force to use a plunger (also makes literally 0 sense, its a chemistry item??) or an rcd (why exactly?), thats making it harder to fix yourself and if you expect robotics to be fixing this, harder for them as they need to go grab random fuckshit items to fix augments/synthetic

> Thats just wrong, actually, I mean-they can get burn damage for a reason. Pressure proof? Yes. Temperature immune? No, if you go out to space your bodytemp shifts.

androids, which are synthetic, are fully temperature immune, just because temp can change doesnt mean anything when they're immune to it, of course they can still take damage from burn because balance, eg why can lasers break windows????

**Comment by nikothedude** [comment]
> serxule words

firstly, I just removed the knockdown and itemdrop because yes, they were too much

The nausea has been heavily restricted, and I'm working on it to make it a bit less dumb - but the shake and dizziness, in testing, has proven to be manageable even at T3, albiet annoying and disrupting - which is the point. THey dont make combat impossible, just more difficult and inconsistant. Ill upload a video eventually actually showing what it looks like.

As for the idea that I'm requiring you to have specific items... no, not at all, if you read the code you'd see that most wounds have only a few steps, and require a TYPE of tool to complete it, which can be HIGHLY varied and can be substituted a lot. 
T3 blunt, for example: Fix superstructure is the first step. You either: Heat the metal with a welder then: Aggrograb, plunger, or hit it a lot, or use an RCD. Then, you secure internals by: Either screwdrivering/wrenching it (if youre a robo/engi/qualified), or you can use bone gel to harden it. This step could probably use another ghetto alternative. Then, you weld it closed with a welder or cautery.
For slash/pierce: Wirecutters can be replaced with a retractor, wires can be replaced by a suture.
For burn: Literally just a fire extinguisher and some gauze, or cryogenics.

No, you aren't locked into using a specific tool. Theres a LOT of ways to fix these wounds. The worse offender is slash/pierce, which doesnt have a way to fix it if you just spawned, and I'm thinking of ways to fix this - but then again, flesh slash and pierce dont have these either - and as someone who is augged, you probably 1. DIDNT just spawn, or 2. already have healing tools. Like wires.

Bone gel works because it works on bone wounds. It is a bit of gel that goes on things, then hardens. It goes onto your internals, then hardens, securing them in place and stopping them from shaking. It is a good tradeoff for not having the proper tools or circumstances to secure your internals - time and damage for guaranteed success.

Also, fair point, didnt really think about how temp immune androids wouldnt really caer about the downsides of spacing yourself to cool yourself. Not sure what to do about that.

**Comment by Serxule** [comment]
> As for the idea that I'm requiring you to have specific items... no, not at all, if you read the code you'd see that most wounds have only a few steps, and require a TYPE of tool to complete it, which can be HIGHLY varied and can be substituted a lot. T3 blunt, for example: Fix superstructure is the first step. You either: Heat the metal with a welder then: Aggrograb, plunger, or hit it a lot, or use an RCD. 

this was really not clear that these were multiple solutions
> Bone gel works because it works on bone wounds. It is a bit of gel that goes on things, then hardens. It goes onto your internals, then hardens, securing them in place and stopping them from shaking. It is a good tradeoff for not having the proper tools or circumstances to secure your internals - time and damage for guaranteed success.

afaik bone gel works by literally melting the bone and then solidifying it to join the bone back together (hence why it causes damage and "you can feel your bones reforming"), it just being a gel would work i guess

**Comment by nikothedude** [comment]
Oh, I see that about the gel. I still want to use it though, we can just say it doesn't corrode medal as much as it reforms bones or w/e.

**Comment by nikothedude** [comment]
Oops I'll remove that later

**Comment by Comxy** [comment]
I suggest adding an ability for slime people to heal all their wounds at the cost of some of their slime jelly like how they can regenerate limbs

**Comment by Ryll-Ryll** [comment]
Personally I would feel a lot better if the synth wounds were moved to a different PR that you could ping me about on discord, and this PR would focus solely on the code improvements

**Comment by Pinkufeck** [comment]
I don't know if buffing EMPs against synthetic limbs is a good move ? 
You use the case of an android to talk about damage, i would just like to remind that getting EMP'd will knockdown an android for 20 seconds, will paralyse them for 6, and their arm will be disabled for an ammount of time (couldn't find it in #74570)
EMPs already will fully disable limbs for  10 seconds minimum i don't know if buffing the damage so won't just make them absolutly inviable at all ?

**Comment by nikothedude** [comment]
> I don't know if buffing EMPs against synthetic limbs is a good move ? You use the case of an android to talk about damage, i would just like to remind that getting EMP'd will knockdown an android for 20 seconds, will paralyse them for 6, and their arm will be disabled for an ammount of time (couldn't find it in #74570) EMPs already will fully disable limbs for 10 seconds minimum i don't know if buffing the damage so won't just make them absolutly inviable at all ?

dude emps got nerfed what

**Comment by nikothedude** [comment]
This PR has been atomized into https://github.com/tgstation/tgstation/pull/77920, discuss the wounds there

**Comment by nikothedude** [comment]
Not opening until I figure out if I should keep cyborg limbs and prosthetics dislocatable

**Comment by nikothedude** [comment]
Not sure why CI failed, I doubt it has anything to do with thsi PR

File: code/datums/wounds/_wounds.dm [hunk 1]
```diff
@@ -171,12 +169,32 @@
 
 		victim.visible_message(msg, span_userdanger("Your [limb.plaintext_zone] [occur_text]!"), vision_distance = vis_dist)
 		if(sound_effect)
-			playsound(L.owner, sound_effect, 70 + 20 * severity, TRUE)
+			playsound(L.owner, sound_effect, sound_volume + (20 * severity), TRUE)
```
Comment (Ryll-Ryll) [inline code/datums/wounds/_wounds.dm:172]: You can get rid of the 20 * severity if this is a var

File: code/datums/wounds/_wounds.dm [hunk 2]
```diff
@@ -445,7 +494,21 @@
 		if(WOUND_SEVERITY_CRITICAL)
 			return "Critical"
 
+
+/// Returns false if our limb is the head or chest, true otherwise.
+/datum/wound/proc/limb_unimportant()
```
Comment (Ryll-Ryll) [inline code/datums/wounds/_wounds.dm:499]: Important in what sense? Can you expand on this?

File: code/datums/wounds/loss.dm [hunk 3]
```diff
@@ -11,9 +17,11 @@
 	wound_flags = null
 	already_scarred = TRUE // We manually assign scars for dismembers through endround missing limbs and aheals
 
+	var/loss_wound_type
```
Comment (Ryll-Ryll) [inline code/datums/wounds/loss.dm:20]: Doc

File: code/datums/wounds/slash.dm [hunk 4]
```diff
@@ -108,7 +121,11 @@
 	if(clot_rate < 0)
 		return BLOOD_FLOW_INCREASING
 
-/datum/wound/slash/handle_process(seconds_per_tick, times_fired)
+/datum/wound/slash/flesh/handle_process(seconds_per_tick, times_fired)
+
+	if (!victim || IS_IN_STASIS(victim))
```
Comment (Ryll-Ryll) [inline code/datums/wounds/slash.dm:126]: I know there's also /datum/wound/proc/on_stasis but can you double check to make sure reducing bloodflow by sutures/cauteries/whatever while someone is on stasis still demotes their wound properly? I recall that being an issue at one point

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 5]
```diff
@@ -186,6 +190,13 @@
 	/// List of the above datums which have actually been instantiated, managed automatically
 	var/list/feature_offsets = list()
 
+	/// In the case we dont have dismemberable features, or literally cant get wounds, we will use this percent to determine when we can be dismembered. Compared to our ABSOLUTE maximum.
+	var/hp_percent_to_dismemberable = 0.8
+	/// If false, no wound that can be applied to us can mangle our flesh. Used for determining if we should use [hp_percent_to_dismemberable] instead of normal dismemberment.
```
Comment (Ryll-Ryll) [inline code/modules/surgery/bodyparts/_bodyparts.dm:195]: Can you be a little more concrete in naming these two?

**Review by Ryll-Ryll — CHANGES_REQUESTED** [review]
This looks very thorough and positive! I'm very much looking forward to seeing where your work ends up. I did figure out halfway through the review that wound_type and wound_series are both in use, if you could document a bit more about what the differences are and what they're used for I'd appreciate it.

Also, just do a quick pass to make sure that wounds can still be healed while someone is on stasis, I feel like the code for that previously was more unintuitive than usual and I wanna make sure I didn't leave any landmines there

File: code/datums/status_effects/debuffs/dizziness.dm [hunk 6]
```diff
@@ -3,6 +3,7 @@
 	tick_interval = 2 SECONDS
 	alert_type = null
 	remove_on_fullheal = TRUE
+	var/applying_dizziness = FALSE
```
[the remark this answers is not in the document]
Comment (nikothedude) [reply code/datums/status_effects/debuffs/dizziness.dm:6]: ah i forgot about this, yeah thats a lil thing that was supposed to be in the synth wounds, forgot to move it over

File: code/__DEFINES/wounds.dm [hunk 7]
```diff
@@ -46,29 +46,76 @@
...
 
-// ~wound global lists
-// list in order of highest severity to lowest
-GLOBAL_LIST_INIT(global_wound_types, list(
-	WOUND_BLUNT = list(/datum/wound/blunt/critical, /datum/wound/blunt/severe, /datum/wound/blunt/moderate),
-	WOUND_SLASH = list(/datum/wound/slash/critical, /datum/wound/slash/severe, /datum/wound/slash/moderate),
-	WOUND_PIERCE = list(/datum/wound/pierce/critical, /datum/wound/pierce/severe, /datum/wound/pierce/moderate),
-	WOUND_BURN = list(/datum/wound/burn/critical, /datum/wound/burn/severe, /datum/wound/burn/moderate),
+// ~biology defines
+// What kind of biology a limb has, and what wounds it can suffer
```
[the remark this answers is not in the document]
Comment (nikothedude) [reply code/__DEFINES/wounds.dm:50]: thanks :^)

File: code/datums/wounds/_wound_static_data.dm [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+		if (required_limb_biostate == null)
+			stack_trace("required_limb_biostate null - please set it! occured on: [src.type]")
+		if (wound_path_to_generate == null)
+			stack_trace("wound_path_to_generate null - please set it! occured on: [src.type]")
+
+// this proc is the primary reason this datum exists - a singleton instance so we can always run this proc even without the wound existing
+/**
+ * Args:
+ * * obj/item/bodypart/limb: The limb we are considering.
+ * * wound_type: The wound type of the wound acquisition attempt. Ex. WOUND_SLASH
```
[the remark this answers is not in the document]
Comment (nikothedude) [reply code/datums/wounds/_wound_static_data.dm:51]: false, wound_type is still it's own thing and is generally the "attack type", or "attack form" that generates wounds

e.g. a blunt attack generates bone, a slash attack generates slashes, etc

File: code/datums/wounds/_wounds.dm (hunk 1 above)
Comment (nikothedude) [reply code/datums/wounds/_wounds.dm:172]: i suppose i could, but then i'd have to go around and change every single sound_volume instance to reflect this

maybe later im a lil low energy tonight

File: code/datums/wounds/slash.dm (hunk 4 above)
Comment (nikothedude) [reply code/datums/wounds/slash.dm:126]: yep it works

**Comment by nikothedude** [comment]
global_all_wound_types was removed out of the simple fact of "it wasnt used anywhere"

File: code/datums/wounds/_wounds.dm (hunk 2 above)
Comment (nikothedude) [reply code/datums/wounds/_wounds.dm]: just renamed it to make more sense

File: code/datums/wounds/_wounds.dm [hunk 9]
```diff
@@ -24,20 +24,28 @@
...
 	/// This sound will be played upon the wound being applied
 	var/sound_effect
+	/// The volume of [sound_effect]
+	var/sound_volume = 70
 
 	/// Either WOUND_SEVERITY_TRIVIAL (meme wounds like stubbed toe), WOUND_SEVERITY_MODERATE, WOUND_SEVERITY_SEVERE, or WOUND_SEVERITY_CRITICAL (or maybe WOUND_SEVERITY_LOSS)
 	var/severity = WOUND_SEVERITY_MODERATE
 	/// The list of wounds it belongs in, WOUND_LIST_BLUNT, WOUND_LIST_SLASH, or WOUND_LIST_BURN
 	var/wound_type
+	/// The series of wounds this is in. Ex. WOUND_SERIES_BLEED_SLASH = avulsions, abrasions...
```
[the remark this answers is not in the document]
Comment (nikothedude) [reply code/datums/wounds/_wounds.dm]: updated docuemntation to say "no"

**Comment by nikothedude** [comment]
i always get so embarassed when i see a dumb mistake in a git diff and rush to fix it (using the wrong variable in set victim)

**Pull request merged**